src/compression.rs

```diff
@@ -1677,18 +1731,41 @@ pub fn compression_step(
 
     let donelist: Vec<FinishedPattern> = shared.crit.lock().deref_mut().donelist.clone();
 
-    let orig_cost = shared.egraph[programs_node].data.inventionless_cost;
+    let orig_cost = shared.egraph[train_programs_node].data.inventionless_cost;
 
     let mut results: Vec<CompressionStepResult> = vec![];
 
     // construct CompressionStepResults and print some info about them)
     println!("Cost before: {}", orig_cost);
+    let mut first_train_cost = None;
+    let mut first_test_cost = None;
     for (i,done) in donelist.iter().enumerate() {
-        let res = CompressionStepResult::new(done.clone(), programs_node, new_inv_name, &mut shared, past_invs);
+        let res = CompressionStepResult::new(
+            done.clone(),
+            train_programs_node,
+            test_programs_node,
+            new_inv_name,
+            &mut shared,
+            past_invs,
+            first_train_cost,
+            first_test_cost);
+
+        // Update initial costs if this was the first step of compression
+        if first_train_cost == None {
+            first_train_cost = Some(res.train_data.initial_cost);
+        }
+        if first_test_cost == None {
+            if let Some(d) = &res.test_data {
+                first_test_cost = Some(d.initial_cost);
+            }
+        }
```

Comment:
As discussed in a previous comment, keeping track of the very first costs (both in train and test) is now the responsibility of the caller of `CompressionStepResult::new`.